code/modules/paperwork/paper.dm

```diff
@@ -1,160 +1,189 @@
+	throw_speed            = 1
+	w_class                = ITEM_SIZE_TINY
+	attack_verb            = list("bapped")
+	material               = /decl/material/solid/paper
+	drop_sound             = 'sound/foley/paperpickup1.ogg'
+	pickup_sound           = 'sound/foley/paperpickup2.ogg'
+	item_flags             = ITEM_FLAG_CAN_TAPE
+	//#TODO: Fonts probably should be stored in the pens or something?
+	var/tmp/deffont        = "Verdana"
+	var/tmp/signfont       = "Times New Roman"
+	var/tmp/crayonfont     = "Comic Sans MS"
+	var/tmp/fancyfont      = "Segoe Script"
+	var/scan_file_type     = /datum/computer_file/data/text
+	var/persist_on_init    = TRUE
+	var/age                = 0
+	var/fields             = 0                      // Amount of user created fields
+	var/free_space         = MAX_PAPER_MESSAGE_LEN
+	var/rigged             = FALSE
+	var/tmp/is_honking     = FALSE
+	var/is_crumpled        = FALSE                  //Whether the paper is currently crumpled
+	var/info                                        // What's actually written on the paper.
+	var/info_links                                  // A different version of the paper which includes html links at fields and EOF (aka the "write" link)
+	var/stamp_text                                  // The (text for the) stamps on the paper.
 	var/list/metadata
+	var/list/applied_stamps                         //List of stamp overlays.
+	var/last_modified_ckey
 
-	var/const/deffont = "Verdana"
-	var/const/signfont = "Times New Roman"
-	var/const/crayonfont = "Comic Sans MS"
-	var/const/fancyfont = "Segoe Script"
-
-	var/scan_file_type = /datum/computer_file/data/text
-
-	var/persist_on_init = TRUE
-
-/obj/item/paper/Initialize(mapload, text, title, list/md = null)
-	. = ..(mapload)
-	set_content(text ? text : info, title)
-	metadata = md
+/obj/item/paper/Initialize(mapload, material_key, var/_text, var/_title, var/list/md = null)
+	. = ..(mapload, material_key)
+	set_content(_text ? _text : info, _title)
+	if(md)
+		LAZYDISTINCTADD(metadata, md) //Merge them
 	if(!mapload && persist_on_init)
 		SSpersistence.track_value(src, /decl/persistence_handler/paper)
 
-/obj/item/paper/create_matter()
-	matter = list(/decl/material/solid/paper = round(SHEET_MATERIAL_AMOUNT * 0.2))
+/**Creates a complete copy of this object */
+/obj/item/paper/proc/Clone()
+	var/obj/item/paper/P = new(loc, , info, name)
+	P.fields             = fields
+	P.last_modified_ckey = last_modified_ckey
+	P.rigged             = rigged
+	P.is_crumpled        = is_crumpled
+	P.stamp_text         = stamp_text
+	P.applied_stamps     = LAZYLEN(applied_stamps)? applied_stamps.Copy() : null
+	P.metadata           = LAZYLEN(metadata)?       metadata.Copy()       : null
+
+	P.set_color(color)
+	P.set_opacity(opacity)
+	P.updateinfolinks()
+	P.update_icon()
+	return P
+
+/obj/item/paper/get_matter_amount_modifier()
+	return 0.2
 
 /obj/item/paper/proc/set_content(text,title)
-	set waitfor = FALSE
+	set waitfor = FALSE //Uh, why?
```

Comment:
Indeed, why?

Reply:
I literally have no ideas why they did this. :/

Reply:
I found why! There's a very cursed proc in the admin paper object :
```
/obj/item/paper/admin/get_signature()
	return input(usr, "Enter the name you wish to sign the paper with (will prompt for multiple entries, in order of entry)", "Signature") as text|null
```
This gets called by parsepencode, which is called by set_content.. 
Seems like someone didn't want to fix their problem and just brushed it under the rug..

```
 code/game/machinery/nuclear_bomb.dm, line 411, column 55:
error: /obj/item/folder/envelope/nuke_instructions/proc/Initialize calls /obj/item/paper/proc/get_signature which has override child proc that sleeps /obj/item/paper/admin/proc/get_signature
- code/game/atoms_init.dm:34:2: SpacemanDMM_should_not_sleep set here
- 414:3: /obj/item/paper/proc/set_content() called here
- code/modules/paperwork/paper.dm:97:9: /obj/item/paper/proc/parsepencode() called here
- code/modules/paperwork/paper.dm:252:65: /obj/item/paper/proc/get_signature() called here
- code/modules/paperwork/adminpaper.dm:160:9: input() called here

code/modules/paperwork/adminpaper.dm, line 22, column 33:
error: /obj/item/paper/admin/proc/Initialize calls /obj/item/paper/proc/get_signature which has override child proc that sleeps /obj/item/paper/admin/proc/get_signature
- code/game/atoms_init.dm:34:2: SpacemanDMM_should_not_sleep set here
- 23:6: /obj/item/paper/proc/Initialize() called here
- code/modules/paperwork/paper.dm:67:2: /obj/item/paper/proc/set_content() called here
- code/modules/paperwork/paper.dm:97:9: /obj/item/paper/proc/parsepencode() called here
- code/modules/paperwork/paper.dm:252:65: /obj/item/paper/proc/get_signature() called here
- 160:9: input() called here

code/modules/paperwork/paper.dm, line 65, column 27:
error: /obj/item/paper/proc/Initialize calls /obj/item/paper/proc/get_signature which has override child proc that sleeps /obj/item/paper/admin/proc/get_signature
- code/game/atoms_init.dm:34:2: SpacemanDMM_should_not_sleep set here
- 67:2: /obj/item/paper/proc/set_content() called here
- 97:9: /obj/item/paper/proc/parsepencode() called here
- 252:65: /obj/item/paper/proc/get_signature() called here
- code/modules/paperwork/adminpaper.dm:160:9: input() called here
```

Reply:
lol, I stg that better not be my code XD